plugins/chat/spec/support/chat_service_matcher.rb

```diff
@@ -23,6 +23,9 @@ def failure_message
           else
             "expected the service to fail but it succeeded."
           end
+
+        result[step].merge("spec.fail_step.unexpected_result" => true) if !step_failed?
```

Comment:
I already played a lot with this code and it would be simpler to apply this each time `failure_message` is called because if it’s called, then we’re in a state we didn’t expect :slightly_smiling_face: 
And we should use it in `failure_message_when_negated`. We need to update the inspector specs too.